src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php

```diff
@@ -70,6 +83,58 @@ public function supports($resource, string $type = null)
 
         return 'php' === $type;
     }
+
+    /**
+     * Resolve the parameters to the $callback and execute it.
+     */
+    private function executeCallback(callable $callback, ContainerConfigurator $containerConfigurator, string $path)
+    {
+        if (!$callback instanceof \Closure) {
+            $callback = \Closure::fromCallable($callback);
+        }
+
+        $arguments = [];
+        $configBuilders = [];
+        $parameters = (new \ReflectionFunction($callback))->getParameters();
+        foreach ($parameters as $parameter) {
+            $reflectionType = $parameter->getType();
+            if (!$reflectionType instanceof \ReflectionNamedType) {
+                throw new \InvalidArgumentException(sprintf('Could not resolve argument "%s" for "%s".', $parameter->getName(), $path));
+            }
+            $type = $reflectionType->getName();
+
+            switch ($type) {
+                case ContainerConfigurator::class:
+                    $arguments[] = $containerConfigurator;
+                    break;
+                case ContainerBuilder::class:
+                case Container::class:
+                case ContainerInterface::class:
+                case \Psr\Container\ContainerInterface::class:
```

Comment:
Not sure about this one and the 2 previous ones: the ContainerBuilder should better not be used as a container at this stage.

Reply:
Thank you. 

I wanted to keep BC. There are users that trust us to provide the following arguments: 

```php
$callback(new ContainerConfigurator(/* .. */), $this->container, $this);
```

So using `\Psr\Container\ContainerInterface` is technically correct. Im more than happy to remove these lines if you still think that is valid.

Reply:
I think `$this->container` is really always a `ContainerBuilder` yes. We should not support the other types.